### PR TITLE
Course Routes

### LINKED ISSUE #8: Create Course
API route to allow users to create a course. Any user can create a course and will be assigned as an instructor by default.

- [ ] Store the creator in the course table.
- [ ] Generate a join code for the course.

### LINKED ISSUE #9: Join Course
API route for a user to join an existing course.

### PR DESCRIPTION
**Pull request by rahulshamkuwar**
Closes #8  and closes #9

### CODE REVIEW

**Comment by rahulshamkuwar** [comment]
@sspathare97 Tbh I feel like the test cases I'm coming up with are superficial and pointless... Could you point me in the right direction for what I could implement for both `courses.service.spec.ts` and `courses.service.spec.ts`?

File: src/auth/auth.controller.ts [hunk 1]
```diff
@@ -15,13 +15,13 @@ export class AuthController {
   @Public()
   @Post("login")
   async login(@Body() loginDto: LoginUserDto) {
-    return this.authService.login(loginDto);
+    return await this.authService.login(loginDto);
```
Comment (sspathare97) [inline src/auth/auth.controller.ts:18]: Why await here?

**Review by sspathare97 — DISMISSED** [review]

**Comment by sspathare97** [comment]
> @sspathare97 Tbh I feel like the test cases I'm coming up with are superficial and pointless... Could you point me in the right direction for what I could implement for both `courses.service.spec.ts` and `courses.service.spec.ts`?

They may seem trivial now but wait until you make breaking changes and the tests break as they should. Tests are a good way to define the expected behavior. It just so happens that the tests you wrote expect trivial CRUD behavior.

File: src/auth/auth.controller.ts (hunk 1 above)
Comment (rahulshamkuwar) [reply src/auth/auth.controller.ts:18]: I was testing something and forgot to remove that it seems.

**Pull request merged**